types/styled-components/test/index.tsx

```diff
@@ -531,11 +539,22 @@ const WithComponentFirstStyledB = WithComponentFirstStyledA.withComponent(
     WithComponentCompB,
 );
 
+const WithComponentFirstStyledANew = styled(WithComponentStyledA).attrs(props => ({ a: 1 }))``;
+
 const test = () => [
     <WithComponentFirstStyledA color={'black'} />,
     <WithComponentFirstStyledB b={2} color={'black'} />,
+    <WithComponentFirstStyledANew color={'black'} />,
 ];
 
+const WithComponentRequired = styled((props: { to: string }) => <a href={props.to}/>)``;
+<WithComponentRequired href=''/>; // $ExpectError
+<WithComponentRequired to=''/>;
+
+const WithComponentRequired2 = WithComponentRequired.withComponent('a');
+<WithComponentRequired2 href=''/>;
+<WithComponentRequired2 to=''/>; // $ExpectError
```

Comment:
These are the tests I fixed with https://github.com/DefinitelyTyped/DefinitelyTyped/pull/30467/files#r232676129